### ISSUE TITLE
Connection should be created in `Hulaaki.Client.handle_call({:connect, _, _}, _, _)`, not `Hulaaki.Client.start_link`

### ISSUE THREAD

**Issue by zoltanmaric** [suvash/hulaaki#41]
Hi Suvash!

Thanks for creating this library.

I've run into an issue, which I believe can be easily fixed. Say I have the simplest possible client:

```
defmodule SampleClient do
  use Hulaaki.Client
end
```

Now if i try to connect to a server which is not reachable:
```
options = [client_id: "some-name-7490", host: "localhost", port: 8883, ssl: true]
{:ok, pid} = SampleClient.start_link(%{})
SampleClient.connect(pid, options)
```

the calling process immediately dies. This is because the connection process is started and linked _in_ the `start_link` function of `Hulaaki.Client`. There's no easy way to connect the client and monitor the connection because in this case the processes are linked and the `Connection` process dies in the same function call.

I believe the solution would be to have the `Connection` process be linked to the `Client` process, instead of linking it to the caller of `Client.connect`. That way, I could start up the `Client`, unlink it and monitor it (you could get extra points for also providing a `start` function in the `Hulaaki.Client` which doesn't automatically link the `Client` to the caller), and then I could recover from a failed call to `connect` by handling the `:DOWN` message in the caller `GenServer`.

Effectively, this change would boil down to moving the following line (currently line 21 of `lib/hulaaki/client.ex`
```
{:ok, conn_pid} = Connection.start_link(pid)
```
from the `start_link` function into the `handle_call` callback for the `:connect` message.

Let me know if that was clear enough and what you think about the suggestion. I'll be happy to contribute with a pull request.

Cheers!

**Comment by suvash** [comment]
Hi there. Thanks for opening up the issue. I've def. been aware of this 'problem' before, and had the same idea as you had to try and fix it. Can't remember why I didn't go for the approach back then.

Give me some days to think about this. I'm in the process of introducing optional Supervisors and some clean up in a month or two.
In the meantime, feel free to submit a PR (with tests). That'll def. help me when it comes to cleanup and improvements for future release.

**Comment by zoltanmaric** [comment]
Thanks for your reply. This week I'm a bit busy, but I can likely work on it on the weekend, so I'll have a PR for you then.

**Comment by suvash** [comment]
This will be resolved once the work on #47 stabilizes and gets merged.
